src/PolynomialRegression2D.ts

```diff
@@ -1,165 +1,139 @@
+import { DataXY, PointXY, type NumberArray } from 'cheminfo-types';
 import { Matrix, SVD } from 'ml-matrix';
-import BaseRegression from 'ml-regression-base';
+import { maybeToPrecision } from 'ml-regression-base';
 
-const defaultOptions = {
-  order: 2,
-};
+import BaseRegression2D, { checkArrayLength } from './BaseRegression2D';
+
+export interface PolynomialRegression2DOptions {
+  /**
+   *degree of the polynomial regression.
+   * @default 2
+   */
+  order?: number;
+}
+
+interface Score {
+  r: number;
+  r2: number;
+  chi2: number;
+  rmsd: number;
+}
 // Implements the Kernel ridge regression algorithm.
 // http://www.ics.uci.edu/~welling/classnotes/papers_class/Kernel-Ridge.pdf
-export class PolynomialRegression2D extends BaseRegression {
+export class PolynomialRegression2D extends BaseRegression2D {
+  order: number;
+  coefficients: Matrix;
+  score: Score;
   /**
    * Constructor for the 2D polynomial fitting
    *
-   * @param inputs
-   * @param outputs
-   * @param options
+   * @param inputs - independent or explanatory variable
+   * @param outputs - dependent or response variable`
    * @constructor
    */
-  constructor(inputs, outputs, options = {}) {
+  constructor(
+    inputs: DataXY,
+    outputs: NumberArray,
+    options: PolynomialRegression2DOptions = {},
+  ) {
     super();
+    // @ts-expect-error internal use only
     if (inputs === true) {
-      // reloading model
+      // @ts-expect-error internal use only
```

Comment:
@targos In order to reload the regression result we use this method that does not seem that nice to me (need to add ts-expect-error).

Could `static load` rather then create a `new PolynomialRegression2D` change the prototype of the object or another way ?

Reply:
We do it like that in other packages. I don't think it's a good idea to make hacks around prototypes.